export_contact_lists.py

```diff
@@ -0,0 +1,115 @@
+    with open(pipeline_configuration_file_path) as f:
+        pipeline_configuration = PipelineConfiguration.from_configuration_file(f)
+    Logger.set_project_name(pipeline_configuration.pipeline_name)
+    log.debug(f"Pipeline name is {pipeline_configuration.pipeline_name}")
+
+    log.info("Downloading Firestore UUID Table credentials...")
+    firestore_uuid_table_credentials = json.loads(google_cloud_utils.download_blob_to_string(
+        google_cloud_credentials_file_path,
+        pipeline_configuration.phone_number_uuid_table.firebase_credentials_file_url
+    ))
+
+    phone_number_uuid_table = FirestoreUuidTable(
+        pipeline_configuration.phone_number_uuid_table.table_name,
+        firestore_uuid_table_credentials,
+        "avf-phone-uuid-"
+    )
+    log.info("Initialised the Firestore UUID table")
+
+    uuids = set()
+    district_counts = {district: 0 for district in TARGET_DISTRICTS}
+    for path in traced_data_paths:
+        # Load the traced data
+        log.info(f"Loading previous traced data from file '{path}'...")
+        with open(path) as f:
+            data = TracedDataJsonIO.import_jsonl_to_traced_data_iterable(f)
+        log.info(f"Loaded {len(data)} traced data objects")
+
+        # Search the TracedData for contacts from one of the relevant locations
+        log.info(f"Searching for participants from the target districts ({TARGET_DISTRICTS})...")
+        file_uuids = set()
+        file_district_counts = {district: 0 for district in TARGET_DISTRICTS}
+        for td in data:
+            if td["district_coded"] == Codes.STOP:
+                continue
+
+            district = CodeSchemes.SOMALIA_DISTRICT.get_code_with_code_id(td["district_coded"]["CodeID"]).string_value
+            if district in TARGET_DISTRICTS:
+                if td["uid"] not in file_uuids:
+                    file_district_counts[district] += 1
+                    file_uuids.add(td["uid"])
+                if td["uid"] not in uuids:
+                    district_counts[district] += 1
+                    uuids.add(td["uid"])
+        log.info(f"Found {len(file_uuids)} contacts in the target locations "
+                 f"(per-district counts: {file_district_counts})")
+        log.info(f"Running total: {len(uuids)} (per-district counts: {district_counts})")
+
+    # Convert the uuids to phone numbers
+    log.info(f"Converting {len(uuids)} uuids to phone numbers...")
+    uuid_phone_number_lut = phone_number_uuid_table.uuid_to_data_batch(uuids)
+    phone_numbers = set()
+    skipped_uuids = set()
+    for uuid in uuids:
+        # Some uuids are no longer re-identifiable due to a uuid table consistency issue between OCHA and WorldBank-PLR
+        if uuid in uuid_phone_number_lut:
+            phone_numbers.add(f"+{uuid_phone_number_lut[uuid]}")
+        else:
+            skipped_uuids.add(uuid)
+    log.info(f"Successfully converted {len(phone_numbers)} uuids to phone numbers. "
```

Comment:
maybe separate this into two; log.info for success and lo.warning for unable to re-identify

Reply:
good point, will do